### PR TITLE
feat(components/container) : add filter on availables display mode on list component

### PR DESCRIPTION
**Pull request by talend-sromain**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

All list mode are displayed to the user.

**What is the new behavior?**

We can filter if we want fewer render modes

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://192.talend.surge.sh)

File: packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js [hunk 1]
```diff
@@ -58,6 +77,7 @@ function SelectDisplayMode({ id, mode, onChange }) {
 SelectDisplayMode.propTypes = {
 	id: PropTypes.string,
 	mode: PropTypes.string,
+	availableModes: PropTypes.arrayOf(PropTypes.string),
```
Comment (talend-glorieux) [inline packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js]: Why not call it displayModes directly?

File: packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js [hunk 2]
```diff
@@ -1,55 +1,74 @@
...
 	const selected = mode || 'table';
 	const displayIcon = (<Icon name={getIcon(selected)} />);
+
 	function onChangeMode(value, event) {
 		return onChange(event, value);
 	}
 
+	function getAvailablesOptions(modes) {
+		if (modes) {
+			return _intersection(modes, options);
```
Comment (talend-sromain) [inline packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js]: as we have lodash on dependencies, why not use it ?
i don't think too it's necessary here btw, just found it cleaner.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://192.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://192.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://192.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://192.talend.surge.sh)

File: packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js [hunk 3]
```diff
@@ -1,55 +1,68 @@
 import React, { PropTypes } from 'react';
 import { Nav, NavDropdown, MenuItem } from 'react-bootstrap';
 import uuid from 'uuid';
+import _intersection from 'lodash/intersection';
```
Comment (jare-talend) [inline packages/components/src/List/Toolbar/SelectDisplayMode/SelectDisplayMode.component.js]: Can you remove this?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://192.talend.surge.sh)

**Pull request merged**